src/layer/marker/Marker.js

```diff
@@ -99,7 +99,9 @@ export const Marker = Layer.extend({
 
 		// @option autoPanSpeed: Number = 10
 		// Number of pixels the map should pan by.
-		autoPanSpeed: 10
+		autoPanSpeed: 10,
+
+		color: null

```

Comment:
The approach is interesting, but I don't like the semantics of this new `L.Marker` option. It works for the default case, but if a user puts an image of a red circle as the icon, then it breaks.

We might want to consider breaking off `L.Icon` from `L.Marker` and do some kind of `L.TintMarker`. Or have the source image to be greyscale and apply a colour to it (not possible with CSS as it seems).